pkg/datapath/iptables/iptables.go

```diff
@@ -433,6 +448,19 @@ func (m *IptablesManager) RemoveRules() {
 	}
 
 	for _, c := range ciliumChains {
+		// do not remove feeder for chains that are set to be disabled
+		skipFeeder := false
+		for _, disabledChain := range option.Config.DisableIptablesFeederRules {
+			if strings.ToLower(c.hook) == strings.ToLower(disabledChain) {
+				log.Infof("Skipping the removal of feeder for chain %s", c.hook)
+				skipFeeder = true
+				break
+			}
+		}
+		if skipFeeder {
+			continue
```

Comment:
Similar to my feedback above, this particular piece of logic seems like it is not very targeted. It will avoid calling `c.remove()` which flushes the cilium chain like `CILIUM_FORWARD`. Maybe I'm missing it right now but I don't think this is responsible for feeder rules at all.

I *think* we can just drop this section entirely. Did you test and find that it is necessary?

Reply:
I'm gonna try without then 😄 but yep I think we can drop it